### PR TITLE
Remove SingleDataSegmentAnnouncer in favor of BatchDataSegmentAnnouncer

### PR DESCRIPTION
**Pull request by guobingkun**
Both Single and Batch Data Segment Announcers exist for backwards compatible reason for a time.

### CODE REVIEW

**Comment by fjy** [comment]
:+1: I forgot to remove this

File: docs/content/configuration/index.md [hunk 1]
```diff
@@ -277,11 +277,7 @@ Data segment announcers are used to announce segments.
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`druid.announcer.type`|Choices: legacy or batch. The type of data segment announcer to use.|batch|
-
-##### Single Data Segment Announcer
-
-In legacy Druid, each segment served by a node would be announced as an individual Znode.
+|`druid.announcer.type`|Choices: batch. The type of data segment announcer to use.|batch|
```
Comment (himanshug) [inline docs/content/configuration/index.md]: why document the property when there is no choice. I believe this configuration should be removed and value should be hard coded to "batch" . also, this PR introduces backward incompatibility for someone having set the property to "legacy" , so a release note update would be required too.

Comment (guobingkun) [inline docs/content/configuration/index.md]: Agree. Removed ```druid.announcer.type``` property from doc.

**Comment by nishantmonu51** [comment]
:+1:

**Pull request merged**